File: frontend1/lib/models/user_stats.dart
class UserStats {
  final String username;
  final int totalSales;
  final double totalRevenue;
  final double averageSaleAmount;
  final int customersServed;
  final List<UserSalesChartData> salesByDate;
  final List<UserTopProduct> topProducts;

  UserStats({
    required this.username,
    required this.totalSales,
    required this.totalRevenue,
    required this.averageSaleAmount,
    required this.customersServed,
    required this.salesByDate,
    required this.topProducts,
  });

  factory UserStats.fromJson(Map<String, dynamic> json) {
    return UserStats(
      username: json['username'] ?? '',
      totalSales: json['total_sales'] ?? 0,
      totalRevenue: (json['total_revenue'] as num?)?.toDouble() ?? 0.0,
      averageSaleAmount: (json['average_sale_amount'] as num?)?.toDouble() ?? 0.0,
      customersServed: json['customers_served'] ?? 0,
      salesByDate: (json['sales_by_date'] as List?)
              ?.map((e) => UserSalesChartData.fromJson(e))
              .toList() ??
          [],
      topProducts: (json['top_products'] as List?)
              ?.map((e) => UserTopProduct.fromJson(e))
              .toList() ??
          [],
    );
  }
}

class UserSalesChartData {
  final String date;
  final double revenue;

  UserSalesChartData({required this.date, required this.revenue});

  factory UserSalesChartData.fromJson(Map<String, dynamic> json) {
    return UserSalesChartData(
      date: json['date'] ?? '',
      revenue: (json['revenue'] as num?)?.toDouble() ?? 0.0,
    );
  }
}

class UserTopProduct {
  final int medicineId;
  final String medicineName;
  final String medicineCode;
  final int quantitySold;
  final double revenueGenerated;

  UserTopProduct({
    required this.medicineId,
    required this.medicineName,
    required this.medicineCode,
    required this.quantitySold,
    required this.revenueGenerated,
  });

  factory UserTopProduct.fromJson(Map<String, dynamic> json) {
    return UserTopProduct(
      medicineId: json['medicine_id'] ?? 0,
      medicineName: json['medicine_name'] ?? '',
      medicineCode: json['medicine_code'] ?? '',
      quantitySold: json['quantity_sold'] ?? 0,
      revenueGenerated: (json['revenue_generated'] as num?)?.toDouble() ?? 0.0,
    );
  }
}

File: deploy_vercel/backend/app/schemas/user_stats.py
"""
User statistics schemas for performance tracking.
"""

from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class TopProductSold(BaseModel):
    """Top product sold by a user."""
    medicine_id: int
    medicine_name: str
    medicine_code: str
    quantity_sold: int
    revenue_generated: float


class UserSalesStats(BaseModel):
    """Sales statistics for a specific user."""
    user_id: int
    username: str
    total_sales: int
    total_revenue: float
    average_sale_amount: float
    customers_served: int
    top_products: List[TopProductSold]
    sales_by_date: List[dict]  # For chart data


class UserPerformance(BaseModel):
    """User performance comparison."""
    user_id: int
    username: str
    role: str
    is_active: bool
    total_sales: int
    total_revenue: float
    average_sale_amount: float
    rank: Optional[int] = None


class UserStatusUpdate(BaseModel):
    """Schema for updating user status."""
    is_active: bool


class SalesStatsFilter(BaseModel):
    """Filter for sales statistics queries."""
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
